src/service/url-replacements-impl.js

```diff
@@ -68,11 +68,11 @@ function encodeValue(val) {
  * Class to provide variables that pertain to top level AMP window.
  */
 export class GlobalVariableSource extends VariableSource {
-
+  /**
+   * @param {!./ampdoc-impl.AmpDoc} ampdoc
+   */
   constructor(ampdoc) {
-    super();
-    /** @const {!./ampdoc-impl.AmpDoc} */
-    this.ampdoc = ampdoc;
+    super(ampdoc);
```

Comment:
you will also need to update all subclass of `VariableSource`.

Reply:
It seems that `GlobalVariableSource` and `A4AVariableSource` are the only subclasses and they are both updated.